users/permissions.py

```diff
@@ -28,7 +33,10 @@ def has_permission(self, request: HttpRequest, view: Any) -> bool:  # pylint: di
     def has_object_permission(  # pylint: disable=no-self-use
         self, request: HttpRequest, _view: Any, obj: CustomUser
     ) -> bool:
+        logger.info(f"Checking permission for object: {request.user}")
         if request.user.is_authenticated:
+            logger.debug(f"User: {request.user} is authenticated")
             return request.user.user_type == CustomUser.UserType.ADMIN.name or obj == request.user
         else:
+            logger.info("User is not authenticated")
```

Comment:
Please add user ID and object ID and object type.

Reply:
Done